low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -357,3 +381,190 @@ def get_nw_status(self, nw_interface, interface):
+        except DBusException:
+            # TODO add logs
+            return None
+        path_array = properties_iface.Get(SERVICE_IFACE, 'ExecStart')
+        try:
+            command_line_path = str(path_array[0][0])
+        except IndexError:
+            # TODO add logs
+            command_line_path = "NA"
+
+        is_installed = True if command_line_path != "NA" or 'invalid' in properties_iface.Get(
+            UNIT_IFACE, 'UnitFileState') else False
+        uid = str(properties_iface.Get(UNIT_IFACE, 'Id'))
+        if not is_installed:
+            health_status = "NA"
+            health_description = f"{uid} is not installed"
+            recommendation = "NA"
+            specifics = [
+                {
+                    "service_name": uid,
+                    "description": "NA",
+                    "installed": str(is_installed).lower(),
+                    "pid": "NA",
+                    "state": "NA",
+                    "substate": "NA",
+                    "status": "NA",
+                    "license": "NA",
+                    "version": "NA",
+                    "command_line_path": "NA"
+                }
+            ]
+        else:
+            service_description = str(
+                properties_iface.Get(UNIT_IFACE, 'Description'))
+            state = str(properties_iface.Get(UNIT_IFACE, 'ActiveState'))
+            substate = str(properties_iface.Get(UNIT_IFACE, 'SubState'))
+            service_status = 'enabled' if 'disabled' not in properties_iface.Get(
+                UNIT_IFACE, 'UnitFileState') else 'disabled'
+            pid = "NA" if state == "inactive" else str(
+                properties_iface.Get(SERVICE_IFACE, 'ExecMainPID'))
+            version = self.get_service_info_from_rpm(uid, "VERSION")
+            service_license = self.get_service_info_from_rpm(uid, "LICENSE")
+            specifics = [
+                {
+                    "service_name": uid,
+                    "description": service_description,
+                    "installed": str(is_installed).lower(),
+                    "pid": pid,
+                    "state": state,
+                    "substate": substate,
+                    "status": service_status,
+                    "license": service_license,
+                    "version": version,
+                    "command_line_path": command_line_path
+                }
+            ]
+            if service_status == 'enabled' and state == 'active' \
+                    and substate == 'running':
+                health_status = 'OK'
+                health_description = f"{uid} is in good health"
+                recommendation = "NA"
+            else:
+                health_status = substate
+                health_description = f"{uid} is not in good health"
+                recommendation = DEFAULT_RECOMMENDATION
+
+        service_info = self.get_health_template(uid, is_fru=False)
+        self.set_health_data(service_info, health_status,
```

Comment:
set_health_data() will take care of fields like description, recommendation based on health status.
No need to explicitly pass the params

Reply:
`set_health_data()` will take care of only two cases, if status is good or status is not good. I have a 3rd case here if service is not installed. So have to add description and recommendation separately.

Reply:
> health_description = f"{uid} is in/not in good health"
    recommendation = DEFAULT_RECOMMENDATION/"NA"
    
  you are setting these values in your code, set_health_data() also does the same

Reply:
Please check the lines 525-528, this is not covered in set_health_data()
In case of status Not Installed (Which is not equal to OK), set_health_data() will set description as f"{uid} is in/not in good health" and recommendation as DEFAULT_RECOMMENDATION which we don't want.